opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchRequestBuilder.java

```diff
@@ -173,14 +174,25 @@ public void pushDownLimit(Integer limit, Integer offset) {
    * Add highlight to DSL requests.
    * @param field name of the field to highlight
    */
-  public void pushDownHighlight(String field) {
+  public void pushDownHighlight(String field, Map<String, Literal> arguments) {
+    String unquotedField = StringUtils.unquoteText(field);
     if (sourceBuilder.highlighter() != null) {
-      sourceBuilder.highlighter().field(StringUtils.unquoteText(field));
+      sourceBuilder.highlighter().field(unquotedField);
     } else {
       HighlightBuilder highlightBuilder =
-          new HighlightBuilder().field(StringUtils.unquoteText(field));
+          new HighlightBuilder().field(unquotedField);
       sourceBuilder.highlighter(highlightBuilder);
     }
+
+
+    if (arguments.containsKey("pre_tags")) {
+      sourceBuilder.highlighter().fields().get(sourceBuilder.highlighter().fields().size() - 1)
```

Comment:
Can you change this to look-up field highlight by name? It's not obvious that the field highlighter returned by `.get(size() - 1)` is the one for `field`

Reply:
On further investigation I don't think this case is possible and I don't think we should include it. I added a comment for clarity but I couldn't find any meaningful way of testing this and it'd be better to just leave a comment than make a hacky test.